docs/architecture/adr-010-unified-cli-arguments-hermes.md

```diff
@@ -0,0 +1,173 @@
+* Any parameter ending with `_id` should have the `_id` removed from the flag to shorten it. For example the flag for `chain_id` should only be `chain`.
+* Flags which can be shorten and still be meaninful should be shorten. This is done for `connection`, `channel` and `sequence`, which become respectively `conn`, `chan` and `seq`.
+* In cases where there are source and destination parameters for a same object, the flags start with the prefix `--src-` and `--dst-`.
+* In cases where there are undirectional parameters for a same object, the flags end with the suffix `-a` and `-b`.
+
+The following commands are implemented, with the binary name `hermes` omitted:
+
+__Hermes global flags__
+
+* `hermes --config <CONFIG> <COMMAND>`
+
+* `hermes --json <COMMAND>`
+
+__Commands for clients__
+
+* `create client --dst-chain <DST_CHAIN_ID> --src-chain <SRC_CHAIN_ID>`
+    * Optional: `[--clock-drift <CLOCK_DRIFT>] [--trust-threshold <TRUST_THRESHOLD>] [--trusting-period <TRUSTING_PERDIOD>]`
+
+* `update client --chain <DST_CHAIN_ID> --client <DST_CLIENT_ID>`
+    * Optional: `[--target-height <TARGET_HEIGHT>] [--trusted-height <TRUSTED_HEIGHT>]`
+
+* `upgrade client --chain <CHAIN_ID> --client <CLIENT_ID>`
+
+* `upgrade clients --chain <SRC_CHAIN_ID>`
+
+__Create a connection__
+
+* `create connection --chain-a <CHAIN_A_ID> --chain-b <CHAIN_B_ID>`
+    * Optional: `[--delay <DELAY>]`
+
+* `create connection --chain-a <CHAIN_A_ID> --client-a <CLIENT_A> --client-b <CLIENT_B>`
+    * Optional: `[--delay <DELAY>]`
+
+__Create a channel__
+
+* `create channel --chain-a <CHAIN_A> --chain-b <CHAIN_B> --port-a <PORT_A> --port-b <PORT_B>`
+    * Optional: `[--chan-version <VERSION>] [--new-client-conn] [--order <ORDER>]`
+
+* `create channel --chain-a <CHAIN_A> --conn-a <CONNECTION_A> --port-a <PORT_A> --port-b <PORT_B>`
+    * Optional: `[--chan-version <VERSION>] [--new-client-conn] [--order <ORDER>]`
+
+__Commands for keys__
+
+* `keys add --chain <CHAIN_ID> --key-file <KEY_FILE> --mnemonic-file <MNEMONIC_FILE>`
+    * Optional: `[--hd-path <HD_PATH>] [--key-name <KEY_NAME>]`
+
+* `keys balance --chain <CHAIN_ID>`
+    * Optional: `[--key-name <KEY_NAME>]`
+
+* `keys delete --chain <CHAIN_ID> --all`
+
+* `keys delete --chain <CHAIN_ID> --key-name <KEY_NAME>`
+
+* `keys list --chain <CHAIN_ID>`
+
+__Listen__
+
+* `listen --chain <CHAIN_ID>`
+    * Optional: `[--event <EVENT>]`
+
+__Misbehaviour__
+
+* `misbehaviour --chain <CHAIN_ID> --client <CLIENT_ID>`
+
+__Start the relayer in multi-chain mode__
+
+* `start`
+    * Optional: `[--full-scan]`
+
+__Clear objects__
+
+* `clear packets --chain <CHAIN_ID> --port <PORT_ID> --chan <CHANNEL_ID>`
+
+__Queries__
+
+* `query channel client --chain <CHAIN_ID> --port <PORT_ID> --chan <CHANNEL_ID>`
+
+* `query channel end --chain <CHAIN_ID> --port <PORT_ID> --chan <CHANNEL_ID>`
+    * Optional: `[--height <HEIGHT>]`
+
+* `query channel ends --chain <CHAIN_ID> --port <PORT_ID> --chan <CHANNEL_ID>`
```

Comment:
This command returns channel ends, connection and clients. Can we find a better name for it? It also currently lies under `query channel`, which to me indicates at a first glance that we are targeting a particular channel (we're not).

Reply:
Thinking about it more, I think it just needs a more descriptive name than `ends`. How about `query channel full` or `query channel stack`? Basically anything that reflects the fact that we're querying for a channel and everything related to it (i.e. connection and client).

WDYT @ljoss17 @adizere ?